src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -1031,27 +1037,25 @@ async function initializeRuntime(
     } catch (err: any) {
       if (err.code !== "ERR_REQUIRE_ESM") {
         await moduleResolutionDetective(frb, err);
-        return;
+        throw err;
       }
       const modulePath = require.resolve(frb.cwd);
       // Resolve module path to file:// URL. Required for windows support.
       const moduleURL = pathToFileURL(modulePath).href;
       triggerModule = await dynamicImport(moduleURL);
     }
   }
-  if (extensionTriggers) {
-    parsedDefinitions = extensionTriggers;
-  } else {
-    require("../deploy/functions/runtimes/node/extractTriggers")(triggerModule, parsedDefinitions);
+  const maybeTrigger = functionTarget.split(".").reduce((mod, functionTargetPart) => {
+    if (typeof mod === "undefined") {
+      return undefined;
+    } else {
+      return mod[functionTargetPart];
```

Comment:
Isn't this just `mod?.[functionTargetPart]`?

Reply:
ah very nice.